include/slang/binding/Bitstream.h

```diff
@@ -0,0 +1,60 @@
+//------------------------------------------------------------------------------
+//! @file Bitstream.h
+//! @brief Helpers for implementing bit-stream casting and streaming operators
+//
+// File is under the MIT license; see LICENSE for details
+//------------------------------------------------------------------------------
+#pragma once
+
+#include <cstddef>
+
+namespace slang {
+
+class Type;
+class ConstantValue;
+class StreamingConcatenationExpression;
+class Expression;
+class ConstantValue;
+class BindContext;
+class EvalContext;
+class SourceRange;
+
+class Bitstream {
+public:
+    /// Compile-time check that the source and destination types have the same dynamic bit-stream
+    /// sizes.
```

Comment:
Can you add a comment about which types are valid here, since only the ones you've explicitly instantiated in the cpp file will work.

Reply:
Yes, I will add comments. Only two types Type and StreamingConcatenationExpression for the template.